mmdet/models/dense_heads/base_dense_head.py

```diff
@@ -14,10 +17,181 @@ def loss(self, **kwargs):
         """Compute losses of the head."""
         pass
 
-    @abstractmethod
-    def get_bboxes(self, **kwargs):
+    @force_fp32(apply_to=('cls_scores', 'bbox_preds'))
+    def get_bboxes(self,
+                   cls_scores,
+                   bbox_preds,
+                   score_factors=None,
+                   img_metas=None,
+                   rescale=False,
+                   with_nms=True,
+                   cfg=None,
+                   **kwargs):
         """Transform network output for a batch into bbox predictions."""
-        pass
+        assert len(cls_scores) == len(bbox_preds)
+
+        if score_factors is None:
+            # e.g. Retina, FreeAnchor, etc.
+            with_score_factors = False
+        else:
+            # e.g. FCOS, PAA, ATSS, etc.
+            with_score_factors = True
+            assert len(cls_scores) == len(score_factors)
+
+        num_levels = len(cls_scores)
+
+        result_list = []
+        for img_id in range(len(img_metas)):
+            img_meta = img_metas[img_id]
+            cls_score_list = [
+                cls_scores[i][img_id].detach() for i in range(num_levels)
```

Comment:
Maybe we can remove the `.detach()`

Reply:
Currently only need to modify the default parameters